Documentation fails to build corrrectly

**Describe the bug**
```text
Generate XML output for dir /Users/ur20980/src/liboqs/src/sig/
Generate XML output for dir /Users/ur20980/src/liboqs/src/
Generating XML output for the main page
/Users/ur20980/src/liboqs/README.md:17: error: unable to resolve reference to 'linuxmacos' for \ref command
Running plantuml with JAVA...
type lookup cache used 152/65536 hits=476 misses=152
symbol lookup cache used 186/65536 hits=566 misses=186
finished...
Exiting...
FAILED: CMakeFiles/gen_docs /Users/ur20980/src/liboqs/build/CMakeFiles/gen_docs 
cd /Users/ur20980/src/liboqs && /Users/ur20980/src/liboqs/scripts/run_doxygen.sh /opt/local/bin/doxygen /Users/ur20980/src/liboqs/docs/.Doxyfile /Users/ur20980/src/liboqs/build
ninja: build stopped: subcommand failed.
```


**To Reproduce**
Clone and run the build.


**Expected behavior**
Successful generation of docs, of course.

Like
```text
Generating XML output for page options-for-configuring-liboqs-builds
Generate XML output for dir /Users/ur20980/src/liboqs/src/common/
Generate XML output for dir /Users/ur20980/src/liboqs/src/kem/
Generate XML output for dir /Users/ur20980/src/liboqs/src/common/rand/
Generate XML output for dir /Users/ur20980/src/liboqs/src/sig/
Generate XML output for dir /Users/ur20980/src/liboqs/src/
Generating XML output for the main page
Running plantuml with JAVA...
type lookup cache used 152/65536 hits=476 misses=152
symbol lookup cache used 186/65536 hits=566 misses=186
finished...

real	0m0.698s
user	0m0.159s
sys	0m0.108s

Built Doxygen documentation
```


**Fix**
_Note: fix includes updating `.Doxygen` to conform to v1.9.8._

```diff
diff --git a/README.md b/README.md
index 7a51186fe..f9b49615d 100644
--- a/README.md
+++ b/README.md
@@ -14,7 +14,7 @@ liboqs is an open source C library for quantum-safe cryptographic algorithms.
                - [Limitations and Security](#limitations-and-security)
                        - [Platform limitations](#platform-limitations)
        - [Quickstart](#quickstart)
-               - [Linux/macOS](#linuxmacos)
+               - [Linux/macOS](#linux/macOS)
                - [Windows](#windows)
                - [Cross compilation](#cross-compilation)
        - [Documentation](#documentation)
```

and please do
```sh
$ cd docs
$ doxygen -u .Doxyfile
```

--- a/.github/workflows/unix.yml
+++ b/.github/workflows/unix.yml
@@ -58,8 +58,11 @@ jobs:
           cat config.log && \
           cmake -LA .. && \
           ! (grep "uninitialized variable" config.log)
-      - name: Build
+      - name: Build code
         run: ninja
+        working-directory: build
+      - name: Build documentation
+        run: ninja gen_docs
         working-directory: build
 
   linux_intel:

--- a/README.md
+++ b/README.md
@@ -14,7 +14,7 @@ liboqs is an open source C library for quantum-safe cryptographic algorithms.
 		- [Limitations and Security](#limitations-and-security)
 			- [Platform limitations](#platform-limitations)
 	- [Quickstart](#quickstart)
-		- [Linux/macOS](#linuxmacos)
+		- [Linux/macOS](#linux/macOS)
 		- [Windows](#windows)
 		- [Cross compilation](#cross-compilation)
 	- [Documentation](#documentation)

